tests/core/test_index.py

```diff
@@ -362,6 +365,7 @@ def test_index_as_flat_series_date():
                 datetime.date(2017, 1, 2),
                 datetime.date(2018, 2, 3),
             ],
+            dtype=dtype,
```

Comment:
What is the purpose of this?

Reply:
I parameterized the `dtype` for the index in order to assert the expected result for `date_as_object=True` and `date_as_object=False`. Otherwise the test will fail.